File: src/main/java/com/projectswg/holocore/client/SWGProtocol.java
package com.projectswg.holocore.client;

import com.projectswg.common.network.NetBuffer;

import java.nio.ByteBuffer;

class SWGProtocol {
	
	private final HolocoreProtocol holocore;
	
	public SWGProtocol() {
		holocore = new HolocoreProtocol();
	}
	
	public void reset() {
		holocore.reset();
	}
	
	public NetBuffer assemble(byte [] packet) {
		return holocore.assemble(packet);
	}
	
	public boolean addToBuffer(ByteBuffer data) {
		return holocore.addToBuffer(data);
	}
	
	public RawPacket disassemble() {
		byte [] packet = holocore.disassemble();
		if (packet.length < 6)
			return null;
		NetBuffer data = NetBuffer.wrap(packet);
		data.getShort();
		return new RawPacket(data.getInt(), packet);
	}
	
	public boolean hasPacket() {
		return holocore.hasPacket();
	}
	
}

File: src/main/java/com/projectswg/holocore/client/SWGProtocol.kt
package com.projectswg.holocore.client

import com.projectswg.common.network.NetBuffer

import java.nio.ByteBuffer

internal class SWGProtocol {
	
	private val holocore: HolocoreProtocol = HolocoreProtocol()
	
	fun reset() {
		holocore.reset()
	}
	
	fun assemble(packet: ByteArray): NetBuffer {
		return holocore.assemble(packet)
	}
	
	fun addToBuffer(data: ByteArray): Boolean {
		return holocore.addToBuffer(data)
	}
	
	fun addToBuffer(data: ByteArray, offset: Int, length: Int): Boolean {
		return holocore.addToBuffer(data, offset, length)
	}
	
	fun addToBuffer(data: ByteBuffer): Boolean {
		return holocore.addToBuffer(data)
	}
	
	fun disassemble(): RawPacket? {
		val packet = holocore.disassemble()
		if (packet.size < 6)
			return null
		val data = NetBuffer.wrap(packet)
		data.position(2)
		return RawPacket(data.int, packet)
	}
	
	fun hasPacket(): Boolean {
		return holocore.hasPacket()
	}
	
}
